src/control/ioserver/config.go

```diff
@@ -0,0 +1,740 @@
+package ioserver
+
+import (
+	"fmt"
+	"os"
+	"strconv"
+	"strings"
+	"unicode"
+)
+
+type (
+	// ValidationError represents an error found while validating
+	// an option
+	ValidationError struct {
+		Option       string
+		ErrorMessage string
+	}
+
+	// SettableOption defines an interface to be implemented
+	// by types which resolve a configuration option to a
+	// command line argument or an environment variable
+	SettableOption interface {
+		Name() string
+		Set(...interface{}) error
+	}
+
+	setterFunc func(...interface{}) error
+
+	// CmdLineArg represents an option which resolves to a
+	// command line argument (switch or option) passed to
+	// an instance on start
+	CmdLineArg struct {
+		Description string
+		LongOption  string
+		ShortOption string
+		Value       string
+
+		optionName    string
+		isEnabledBool bool
+		validateFunc  func(*CmdLineArg) error
+		setterFunc    setterFunc
+	}
+
+	// EnvVar represents an option which resolves
+	// to an environment variable set in the instance's
+	// environment before starting
+	EnvVar struct {
+		Description string
+		Key         string
+		Value       string
+
+		optionName   string
+		validateFunc func(*EnvVar) error
+		setterFunc   setterFunc
+	}
+
+	// Config represents an instance's configuration
+	// and comprises command line arguments as well
+	// as environment variables
+	Config struct {
+		SystemName               string
+		ShouldReformatSuperblock bool
```

Comment:
I'm interested in why this is here and it's usage.

Reply:
Hmm. I thought i'd removed those fields from the version that got pushed. I'll have to fix that. I was doing more in the ioserver package before I really fleshed out the configuration aspect. Those things belong in the instance harness in the control server package.